sync-core/src/main/java/com/cloudant/sync/replication/Filter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.replication;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <p>Provides the name and parameters for a filter function to be used
+ * when a pull replication calls the source database's {@code _changes}
+ * feed.</p>
+ *
+ * <p>For a filter function that takes no parameters, use this
+ * constructor:</p>
+ *
+ * <pre>
+ * Filter filter = new Filter("filterDoc/filterFunctionName")
+ * </pre>
+ *
+ * <p>For a filter function that requires parameters, use this
+ * constructor:</p>
+ *
+ * <pre>
+ * Map&lt;String, String&gt; params = new HashMap();
+ * map.put("max_age", "10");
+ * map.put("name", "john");
+ * Filter filter = new Filter("doc/filterName", map);
+ * </pre>
+ *
+ * @see <a href="http://couchdb.readthedocs.org/en/1.6.x/replication/intro.html#controlling-which-documents-to-replicate">Controlling documents replicated</a>
+ * @see <a href="http://docs.couchdb.org/en/1.6.x/couchapp/ddocs.html#filter-functions">Filter functions CouchDB docs</a>
+ */
+public class Filter {
```

Comment:
`PullFilter` so it's not too huge a name? It's in the replication package so it's clear that's it replication specific.